x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/experimental_datastream_settings.tsx

```diff
@@ -93,6 +96,11 @@ export const ExperimentDatastreamSettings: React.FunctionComponent<Props> = ({
       });
     }
 
+    // This sets the features on change, will it update data stream itself?
+    // Is this were rollover should happen?
+    // Ideally there would be a "data management" part in Fleet (or outside) where on a change
+    // the policy API could just make a call with the requested change and would get back if it was applied or
+    // not, not having to think about templates, data streams etc.
```

Comment:
This is just the frontend state tracking of the opt-in state for each experimental indexing feature. The `package_policy` API handles actually updating templates here: 

https://github.com/elastic/kibana/blob/86527753a2db307138ea69e0ba04551717a91844/x-pack/plugins/fleet/server/services/package_policy.ts#L182-L183

https://github.com/elastic/kibana/blob/86527753a2db307138ea69e0ba04551717a91844/x-pack/plugins/fleet/server/services/package_policies/experimental_datastream_features.ts#L44-L152

> Is this where rollover should happen

Automatic rollover should happen in the above server-side code. I don't think it'd be too challenging, but there might be caveats of which I am unaware. Issue at https://github.com/elastic/kibana/issues/143448

> Ideally there would be a "data management" part in Fleet (or outside) where on a change the policy API could just make a call with the requested change and would get back if it was applied or not, not having to think about templates, data streams etc.

I'm in agreement here. We've talked about something like a `/api/fleet/customize` API to handle cases like this around pushing custom configuration objects. @joshdover has a great pass at designing this kind of API at https://github.com/elastic/kibana/issues/121118, namely the `Customize API` section feels relevant here.

Reply:
Great to see this is already all started to be tracked. The reason I'm keen on https://github.com/elastic/kibana/issues/143448 to happen is that I think at the moment when a user "toggles" any of the features, the user assumes it is working now but in reality the user has to trigger a rollover first. We already do rollovers on upgrades when needed so I would assume the functionality + error handling mostly already exists.

> /api/fleet/customize

I might go further and not put it into the `fleet` plugin. From my perspective it goes so far, that this is a data management API that should be exposed by Kibana "core" and is only used by Fleet. Need to read up on the full issue and comment there.